docs/operations/clean-metadata-store.md

```diff
@@ -0,0 +1,104 @@
+
+Do not use the metadata store automated cleanup features if you have requirements to retain metadata records. For example, you have compliance requirements to keep audit records. In these cases, you should come up with an alternate method to preserve the audit metadata while freeing up your active metadata store.
+
+## Configure automated metadata cleanup
+Automated cleanup only removes records for deleted datasources. You can configure cleanup on a per-table basis as follows:
+ - `druid.coordinator.kill.*.on` enables cleanup for a particular metadata table.
+ - `druid.coordinator.kill.*.period` defines the frequency in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations) for the cleanup job to check for and delete eligible records.
+ - `druid.coordinator.kill.*.durationToRetain` defines the retention period in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations) after creation that the entity becomes eligible for deletion. For example to retain records for 30 days from their creation date: `P30D`.
+
+ For full configuration details, see [Metadata management](../configuration/index.md#metadata-management).
+
+The following are eligible for automatic cleanup:
+- `druid_audit`: Audit records for deleted datasources become eligible for deletion when the `durationToRetain` time has passed since their creation.
+- `druid_supervisors`: Supervisor records become eligible for deletion when:
+
+    - the supervisor is terminated.
+    - the `durationToRetain` time has passed since their creation.
+- `druid_rules`: Rule records become eligible for deletion when:
+
+    - no segments exist for the datasource.
+    - the `durationToRetain` time has passed since their creation.
+- `compaction`: Compaction configuration records in the records in the `druid_config` table become eligible for deletion when no segments exist for the datasource.
+     >If you already have an extremely large compaction configuration, you may not be able to delete compaction configuration due to size limits with the audit log. In this case you can set `druid.audit.manager.maxPayloadSizeBytes` and `druid.audit.manager.skipNullField` to avoid the auditing issue. See [Audit logging](../configuration/index.md#audit-logging).
+- `druid_datasources`: Datasource records become eligible for deletion when:
+
+    - the supervisor is terminated or does not exist in the `druid_supervisors` table.
+    - the `durationToRetain` time has passed since their creation.
+
+## Automated metadata cleanup example configuration
+Consider a scenario where you have scripts to create and delete supervisors for hundreds of datasources a day. You do not want to fill your metadata store with records for deleted datasources. Datasources tend to persist for only one or two days. Therefore, you want to run a cleanup job that identifies and removes leftover records that are at least four days old. The exception is for audit logs, which you need to retain for 30 days:
+
+```
```

Comment:
There's an additional configuration `druid.coordinator.period.metadataStoreManagementPeriod`. This is the scheduler that runs all the individual clean up jobs. The `druid.coordinator.period.metadataStoreManagementPeriod` must also be set to be less than or equal to the smallest period of all the clean up task. By default it is 1 hour which will work for this example, since all the individual cleanup task period is greater than 1 hour

Reply:
Might also be worth mentioning that you need the `kill` task too. The kill task is required to delete the segments after the datasource is not needed, this is required for compaction and rules to be eligible for cleanup. The kill task can be scheduled automatically to run via configs `druid.coordinator.kill.on`, `druid.coordinator.kill.period`, `druid.coordinator.kill.durationToRetain`, `druid.coordinator.kill.maxSegments`, etc. You can find what each one does in the https://druid.apache.org/docs/latest/configuration/index.html page